src/main/java/org/jabref/gui/EntryTypeView.java

```diff
@@ -0,0 +1,256 @@
+
+            @Override
+            public String toString(IdBasedFetcher object) {
+                return object.getName();
+            }
+
+            @Override
+            public IdBasedFetcher fromString(String string) {
+                return null;
+            }
+        });
+
+        //we set the managed property so that they will only be rendered when they are visble so that the Nodes only take the space when visible
+        //avoids removing and adding from the scence graph
+        bibTexTitlePane.managedProperty().bind(bibTexTitlePane.visibleProperty());
+        ieeeTranTitlePane.managedProperty().bind(ieeeTranTitlePane.visibleProperty());
+        biblatexTitlePane.managedProperty().bind(biblatexTitlePane.visibleProperty());
+        customTitlePane.managedProperty().bind(customTitlePane.visibleProperty());
+    }
+
+    public EntryType getChoice() {
+        return type;
+    }
+
+    private void stopFetching() {
+        if (fetcherWorker.getState() == Worker.State.RUNNING) {
+            fetcherWorker.cancel(true);
+        }
+    }
+
+    private class FetcherWorker extends Task<Optional<BibEntry>> {
+
+        private boolean fetcherException = false;
+        private String fetcherExceptionMessage = "";
+        private IdBasedFetcher fetcher = null;
+        private String searchID = "";
+
+        @Override
+        protected void done() {
+            try {
+                Optional<BibEntry> result = get();
+                if (result.isPresent()) {
+                    final BibEntry bibEntry = result.get();
+                    if ((DuplicateCheck.containsDuplicate(basePanel.getDatabase(), bibEntry, basePanel.getBibDatabaseContext().getMode()).isPresent())) {
+                        //If there are duplicates starts ImportInspectionDialog
+                        final BasePanel panel = basePanel;
+
+                        ImportInspectionDialog diag = new ImportInspectionDialog(basePanel.frame(), panel, Localization.lang("Import"), false);
+                        diag.addEntries(Arrays.asList(bibEntry));
+                        diag.entryListComplete();
+                        diag.setVisible(true);
+                        diag.toFront();
+                    } else {
+                        // Regenerate CiteKey of imported BibEntry
+                        new BibtexKeyGenerator(basePanel.getBibDatabaseContext(), prefs.getBibtexKeyPatternPreferences()).generateAndSetKey(bibEntry);
+                        // Update Timestamps
+                        if (prefs.getTimestampPreferences().includeCreatedTimestamp()) {
+                            bibEntry.setField(prefs.getTimestampPreferences().getTimestampField(), prefs.getTimestampPreferences().now());
+                        }
+                        basePanel.insertEntry(bibEntry);
+                    }
+
+                    close();
+                } else if (searchID.trim().isEmpty()) {
```

Comment:
better disable the generate button if the search field is empty (and use StringUtil.isBlank)